File: cpp/merge/Merger_pdf_docx.cpp
#include <iostream>

#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/IO/ConvertStrategies/PdfFormat.h"
#include "Aspose.PDF.Cpp/PdfFormatConversionOptions.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/DocSaveOptions/DocFormat.h"
#include "Aspose.PDF.Cpp/DocSaveOptions/RecognitionMode.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"

#include "system/string.h"

using namespace System;
using namespace Aspose::Pdf;

void pdf_to_docx()
{
    auto pathSource1 = u"../../TestData/test.pdf";
    auto pathSource2 = u"../../TestData/Second/test.pdf";

    // read pdf file to Aspose Document
    auto firstDoc = MakeObject<Document>(pathSource1);
    auto secondDoc = MakeObject<Document>(pathSource2);

    // create empty pdf document
    auto outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    outputDoc->EnableObjectUnload = true;

    for (int i = 0; i < firstDoc->get_Pages()->get_Count(); i++)
    {
        auto page = firstDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    for (int i = 0; i < secondDoc->get_Pages()->get_Count(); i++)
    {
        auto page = secondDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    var opt1 = MakeObject<DocSaveOptions>();
    // use docx format
    opt1->Format = DocFormat::DocX;
    // make document editable flow and recognize of tables
    opt1->Mode = RecognitionMode::EnhancedFlow;

    outputDoc->Save("test.docx", opt1);
}

File: python/merge/Merger_pdf_docx.py
from aspose.pdf import (
    DocSaveOptions,
    Document
)


def pdf_to_docx():
    path_source1 = "../../TestData/test.pdf"
    path_source2 = "../../TestData/Second/test.pdf"

    # read pdf file to Aspose Document
    first_doc = Document(path_source1)
    second_doc = Document(path_source2)

    # create empty pdf document
    output_doc = Document()

    # set less memory usage with unload instead of fast performance
    output_doc.enable_object_unload = True

    for page in first_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    for page in second_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    opt1 = DocSaveOptions()
    # use docx format
    opt1.format = DocSaveOptions.DocFormat.DOC_X
    # make document editable flow and recognize of tables
    opt1.mode = DocSaveOptions.RecognitionMode.ENHANCED_FLOW

    # save pdf to Microsoft Word docx format
    output_doc.save("Merger_pdf_docx.docx", opt1)
